pymatgen/io/qchem/outputs.py

```diff
@@ -976,6 +977,137 @@ def _read_charges_and_dipoles(self):
+                        }
+                    )
+
+        temp_hexadecapole_moment = read_pattern(
+            self.text,
+            {
+                "key": (
+                    r"\s*Hexadecapole Moments \(Debye\-Ang\^3\)\s+XXXX\s+([\-\.0-9]+)\s+XXXY\s+([\-\.0-9]+)"
+                    r"\s+XXYY\s+([\-\.0-9]+)\s+XYYY\s+([\-\.0-9]+)\s+YYYY\s+([\-\.0-9]+)\s+XXXZ\s+([\-\.0-9]+)"
+                    r"\s+XXYZ\s+([\-\.0-9]+)\s+XYYZ\s+([\-\.0-9]+)\s+YYYZ\s+([\-\.0-9]+)\s+XXZZ\s+([\-\.0-9]+)"
+                    r"\s+XYZZ\s+([\-\.0-9]+)\s+YYZZ\s+([\-\.0-9]+)\s+XZZZ\s+([\-\.0-9]+)\s+YZZZ\s+([\-\.0-9]+)"
+                    r"\s+ZZZZ\s+([\-\.0-9]+)"
+                )
+            },
+        ).get("key")
+        if temp_hexadecapole_moment is not None:
+            if len(temp_hexadecapole_moment) == 1:
+                self.data["multipoles"]["hexadecapole"] = {
+                    "XXXX": float(temp_hexadecapole_moment[0][0]),
+                    "XXXY": float(temp_hexadecapole_moment[0][1]),
+                    "XXYY": float(temp_hexadecapole_moment[0][2]),
+                    "XYYY": float(temp_hexadecapole_moment[0][3]),
+                    "YYYY": float(temp_hexadecapole_moment[0][4]),
+                    "XXXZ": float(temp_hexadecapole_moment[0][5]),
+                    "XXYZ": float(temp_hexadecapole_moment[0][6]),
+                    "XYYZ": float(temp_hexadecapole_moment[0][7]),
+                    "YYYZ": float(temp_hexadecapole_moment[0][8]),
+                    "XXZZ": float(temp_hexadecapole_moment[0][9]),
+                    "XYZZ": float(temp_hexadecapole_moment[0][10]),
+                    "YYZZ": float(temp_hexadecapole_moment[0][11]),
+                    "XZZZ": float(temp_hexadecapole_moment[0][12]),
+                    "YZZZ": float(temp_hexadecapole_moment[0][13]),
+                    "ZZZZ": float(temp_hexadecapole_moment[0][14]),
+                }
+            else:
+                self.data["multipoles"]["hexadecapole"] = list()
+                for hpole in temp_hexadecapole_moment:
+                    self.data["multipoles"]["hexadecapole"].append(
+                        {
+                            "XXXX": float(hpole[0]),
+                            "XXXY": float(hpole[1]),
+                            "XXYY": float(hpole[2]),
+                            "XYYY": float(hpole[3]),
+                            "YYYY": float(hpole[4]),
+                            "XXXZ": float(hpole[5]),
+                            "XXYZ": float(hpole[6]),
+                            "XYYZ": float(hpole[7]),
+                            "YYYZ": float(hpole[8]),
+                            "XXZZ": float(hpole[9]),
+                            "XYZZ": float(hpole[10]),
+                            "YYZZ": float(hpole[11]),
+                            "XZZZ": float(hpole[12]),
+                            "YZZZ": float(hpole[13]),
+                            "ZZZZ": float(hpole[14]),
```

Comment:
This could be made more concise with a list comprehension

```py
keys = ["XXXX", "XXXY", "XXYY", "XYYY", "YYYY", "XXXZ", "XXYZ", "XYYZ", "YYYZ", "XXZZ", "XYZZ", "YYZZ", "XZZZ", "YZZZ", "ZZZZ"]
hexadecapole = [{key: float(hpole[idx]) for idx, key in enumerate(keys)}]
```

Likewise above.